### PR TITLE
Adds mint, a new plant that can be processed into menthol

### PR DESCRIPTION
**Pull request by sneksnek**
## What Does This PR Do
adds the mint plant. its intended purpose is to be grown so it may be processed for the mint chemical, which in turn can be combined with ethanol to create menthol. all mint sprites were crudely drawn by me, and new mint seeds are in the vendor. a new recipe has been added, requiring equal parts mint and ethanol to create menthol

## Why It's Good For The Game
menthol is a fun chem to make healing mixes with, but it's currently very time-consuming to acquire a reasonable amount. this PR gives it a better source, so that doctors, chemists, or botanists may make more use of it if they're feeling creative and/or bored. additionally, mint could be given more uses in cooking in the future

## Images of changes
mint_overview
mint_plants
mint_seeds
mint_leaf
menthol

## Changelog
:cl:
add: Added mint plants and a menthol recipe
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/herbals.dm [hunk 1]
```diff
@@ -53,3 +53,28 @@
...
+	reagents_add = list("mint" = 0.03, "plantmatter" = 0.03)
+
+/obj/item/reagent_containers/food/snacks/grown/mint
+	seed = /obj/item/seeds/mint
+	name = "mint leaves"
+	desc = "Process for mint. No need to experi-mint." //haha
+	icon_state = "mint"
+	tastes = list("mint" = 1)
+	filling_color = "#008000"
+	distill_reagent = "mint"
```
Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/herbals.dm:80]: I'd have it distill into full menthol.

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 2]
```diff
@@ -703,3 +703,11 @@
 
 	if(H.dna.species.bodyflags & HAS_SKIN_COLOR) //take current alien color and darken it slightly
 		H.change_skin_color("#9B7653")
+
+/datum/reagent/mint
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/misc.dm:707]: should probably be a consumable reagent

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 3]
```diff
@@ -703,3 +703,11 @@
...
+	name = "Mint"
+	id = "mint"
+	description = "Processed mint, useful in the production of menthol."
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/misc.dm:710]: `description = "A light green liquid extracted from mint leaves."`

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 4]
```diff
@@ -703,3 +703,11 @@
...
+	reagent_state = LIQUID
+	color = "#F8F8FF"
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/misc.dm:712]: `"#A7EE9F"`

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 5]
```diff
@@ -280,3 +280,11 @@
...
 	min_temp = T0C + 100
 	mix_message = "The solution gently swirls with a metallic sheen."
+
+/datum/chemical_reaction/menthol
+	name = "Menthol"
+	id = "menthol"
+	result = "menthol"
+	required_reagents = list("mint" = 1, "ethanol" = 1)
+	result_amount = 2
+	mix_message = "The mixture precipitates into many large white crystals."
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/medicine.dm:290]: `"Large white crystals precipitate out of the mixture."`

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 6]
```diff
@@ -280,3 +280,11 @@
 	result_amount = 3
 	min_temp = T0C + 100
 	mix_message = "The solution gently swirls with a metallic sheen."
+
+/datum/chemical_reaction/menthol
+	name = "Menthol"
+	id = "menthol"
+	result = "menthol"
+	required_reagents = list("mint" = 1, "ethanol" = 1)
+	result_amount = 2
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/medicine.dm:289]: 1

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/recipes/medicine.dm:289]: `min_temp = T0C + 50`

File: code/modules/hydroponics/grown/herbals.dm [hunk 7]
```diff
@@ -53,3 +53,28 @@
...
+	icon_dead = "mint-dead"
+	reagents_add = list("mint" = 0.03, "plantmatter" = 0.03)
+
+/obj/item/reagent_containers/food/snacks/grown/mint
+	seed = /obj/item/seeds/mint
+	name = "mint leaves"
+	desc = "Process for mint. No need to experi-mint." //haha
+	icon_state = "mint"
+	tastes = list("mint" = 1)
+	filling_color = "#008000"
```
Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/herbals.dm:79]: filling color the same as the reagent.

**Review by Kyep — APPROVED** [review]

**Pull request merged**